hostfactory/host_provider/test/cluster_test.py

```diff
@@ -4,12 +4,18 @@
 from unittest.mock import MagicMock
 from unittest.mock import patch
 
+
+class MockResult:
+    def __init__(self, status, node_list):
+        self.status = status
+        self.node_list = node_list
 class MockNodeMgr:
     
-    def __init__(self, expect=[], expect_new_nodes=0):
+    def __init__(self, expect=[], expected_new_nodes_list=[], expected_result="success"):
```

Comment:
what is the purpose of expected_result in a MockNodeMgr?

Reply:
so basically its result status and list of nodes. I added this for testing the case of NoAllocationSelected